### PR TITLE
[2 of 4] New Signed Out Homepage (new navbar)

### PR DESCRIPTION
**Pull request by wgranger**
**Describe your changes:**
This branch includes changes to the nav bar to be merged into the new-signed-out-homepage branch. 

_Note_: This branch will merge into that other branch and not master.

https://new-navbar-style.pfe-preview.zooniverse.org/

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
This looks really ugly on a project page. The black doesn't fill the width of the viewport.
Screenshot of the new nav bar on a project, showing that it only fills about half the viewport width.

File: app/layout/account-bar.jsx [hunk 1]
```diff
@@ -17,55 +16,57 @@ const DOWN = 40;
...
-  },
+
+    this.handleAccountMenuOpen = this.handleAccountMenuOpen.bind(this);
+    this.navigateMenu = this.navigateMenu.bind(this);
+    this.handleSignOutClick = this.handleSignOutClick.bind(this);
+  }
 
   componentDidMount() {
     addEventListener('locationchange', this.lookUpUnread);
     this.lookUpUnread();
```
Comment (eatyourgreens) [inline app/layout/account-bar.jsx]: For this listener to work, `this.lookUpUnread` should be bound to the component's `this` context in the constructor.

File: app/layout/account-bar.jsx [hunk 2]
```diff
@@ -77,47 +78,88 @@ const AccountBar = React.createClass({
...
-      type: 'logout',
+      type: 'logout'
     });
 
     auth.signOut();
-  },
+  }
+
+  renderMessages() {
+    const mobile = this.state.unread ? <i className="fa fa-envelope fa-fw" /> : <i className="fa fa-envelope-o fa-fw" />;
```
Comment (eatyourgreens) [inline app/layout/account-bar.jsx]: This should have a text label for VoiceOver to read out.

File: app/layout/account-bar.jsx [hunk 3]
```diff
@@ -77,47 +78,88 @@ const AccountBar = React.createClass({
...
+        </Link>
+
         <TriggeredModalForm
           ref="accountMenuButton"
           className="site-nav__modal"
           trigger={
             <span className="site-nav__link">
               <strong>{this.context.user.display_name}</strong>{' '}
-              <Avatar className="site-nav__user-avatar" user={this.context.user} size="2em" />
+              <i className="fa fa-chevron-down" />
```
Comment (eatyourgreens) [inline app/layout/account-bar.jsx]: People might miss having their avatar displayed here 😞

**Comment by srallen** [comment]
Is there really any reason to keep doing the demoted/promoted functionality and styling anymore with the projects with this design? @beckyrother can you share your design thoughts on the navbar changes?

**Comment by eatyourgreens** [comment]
True, that. Also, it's kind of disconcerting that the black navigation changes back to the teal design when you log in. This is one component that's going to have to change globally, for the whole site.

**Comment by beckyrother** [comment]
Can this be a global change? @eatyourgreens is right, it's odd for the navigation to change when you log in. 
@srallen I'm not sure what you mean about demoted/promoted functionality, but my thinking behind this is that a darker bar separate from the content will fade more into the background and be less distracting. I'd like this to be visible all the time, so it replaces the hamburger menu with the motion. In the classify/project section it'll also allow the project's background image and avatar to shine and really obviously differentiate between project navigation and site navigation. Here's an example of that: 
page-dark_png

**Comment by srallen** [comment]
The demoted/promoted is the hamburger menu and that animation (demoted/promoted are the variables and class names that are being used in the code). Thanks for clarifying.

We should go forward with removing that, since it'll make styling and the javascript simpler.

**Comment by eatyourgreens** [comment]
On OSX, the blue focus ring is lost against the teal background, making it difficult to see which item is selected when you are using a keyboard to select from the menu.
User account menu showing blue OSX is barely visible against a teal background
I think any `:hover` styles should also apply to the `:focus` pseudo-class.

**Comment by wgranger** [comment]
This has been restaged with the demoted functionality removed. That functionality hit a good bit of code, but I couldn't find any bugs after removing it from this branch.

**Comment by CKrawczyk** [comment]
The user dropdown is causing horizontal scrolling on my Mac (both chrome and firefox).

screen shot 2017-05-15 at 9 42 54 am

**Comment by wgranger** [comment]
@CKrawczyk Let me know if you're still experiencing the horizontal scrolling issue from above. I'm unable to replicate it on the [staged branch above](https://new-navbar-style.pfe-preview.zooniverse.org/). I thought I took care of that in a previous commit, but I could be missing something.

**Comment by mschwamb** [comment]
On Mac OS when using Firefox, the first item in the account dropdown is always selected (I think - it's got the white box around it) even after you move past it, which is confusing about what's really selected. 

screen shot 2017-05-15 at 11 35 58 am

I didn't experience the horizontal scrolling.

**Comment by mschwamb** [comment]
Maybe not the right place for this... Also if you're updating the navigation bar, can you please consider adding a shortcut to the specific project lab page if you're a project owner or researcher on the project. Having three panoptes projects, I find it a little cumbersome to always have to click on the lab then  click on the list to the project I want. Usually I see something needs editing when I'm at the site on Talk.  See ticket #3363

**Comment by CKrawczyk** [comment]
I am still seeing the scrolling on my computer (I even tried incognito mode).

**Comment by wgranger** [comment]
Ok. A fix might need to be added to the modal-form repo, because this is also happening when I open the main site on my phone.

**Comment by wgranger** [comment]
@eatyourgreens Do you have a suggestion (considering accessibility) to the hover/focus comment made by Meg above? The functionality on this branch is similar to how we currently have it on master.

**Comment by eatyourgreens** [comment]
@wgranger that's there for keyboard accessibility. Check it out on https://www.zooniverse.org.

The first item is selected when you open the menu. Use the up/down cursor keys to move up and down the menu, enter to activate, esc to close the menu and focus the original button.

You should style the menu so that any `:hover` styles also use `:focus`, I think.

**Comment by wgranger** [comment]
The hover and focus styles on the menu are currently shared. I'm mainly curious if there's a way to only focus/hover one item at a time (rather than one item receiving focus while another has hover). I think that was the main issue brought up above.

**Comment by eatyourgreens** [comment]
Oh right. No, hover doesn't focus an item. Only clicking will do that (and then not always depending on the browser.)

OSX usually has a bright blue ring to indicate focus, but if I remember rightly, the original issue was that that ring doesn't show up against the teal background.

Maybe have a chat with @beckyrother about how focus should be indicated here, as compared to mouse hover or touch.

Another option is to leave focus with the original element when the menu opens, then enter the menu with the down arrow. See the search box on twitter.com for an example of this. I can't remember if our components allow for this, though.

**Comment by beckyrother** [comment]
Good point @eatyourgreens. Here's a pdf with what I'm thinking for the hover vs selected:

[nav-signedin-dropdown.pdf](https://github.com/zooniverse/Panoptes-Front-End/files/1019585/nav-signedin-dropdown.pdf)

First page is before you hover over anything else, second shows the hover state. Things to note, and things I've noted from the current PR:
- highlight box goes across the entire dropdown box
- no line between Favorites and Sign Out
- letter-spacing should be 0.15em across the entire nav bar

Let me know if there are any questions. Thanks!

**Comment by wgranger** [comment]
There's currently a [PR](https://github.com/zooniverse/modal-form/pull/31) open in the modal-form repo to fix the issue with the menu being shown offscreen.

**Comment by beckyrother** [comment]
That's great @eatyourgreens, thanks!

**Pull request merged**